src/pybind/mgr/prometheus/module.py

```diff
@@ -1273,6 +1275,9 @@ def respond():
             (server_addr, server_port)
         )
 
+        self.metrics_thread = MetricCollectionThread(_global_instance)
+        self.metrics_thread.start()
```

Comment:
Originally it could but it wasn't passing `make check` tests (it was complaining about a type mismatch `Optional[Module]` vs just `Module` so I had to move the class initialization to `__init__`. Hence, it can't be a local variable anymore.